Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -414,7 +414,7 @@ inline HRESULT PopEndDraw() {
     template <typename Lambda> // Lambda takes the form HRESULT (*)(CGContextRef, ID2D1DeviceContext*)
     HRESULT Draw(_CGCoordinateMode coordinateMode, CGAffineTransform* additionalTransform, Lambda&& drawLambda);
 
-    HRESULT DrawGeometry(_CGCoordinateMode coordinateMode, ID2D1Geometry* pGeometry, CGPathDrawingMode drawMode);
+    HRESULT DrawGeometry(_CGCoordinateMode coordinateMode, ID2D1Geometry* pGeometry, CGPathDrawingMode drawMode, bool isPathGeometry);
```

Comment:
Not sure if I like this.  Since the cases where it is "false" you are creating a geometry just before calling DrawGeometry() can you do the conversion there?  More fundamentally, why do we need this conversion for built-in geometry like Rectangle, seems wasteful?